### DISCUSSION TITLE
Uncaught (in promise) ReferenceError: Cesium is not defined     at vue-cesium.js?v=2048c4fa:23523:27，使用的环境是Vue3+js，初始化的时候更换imagery-provider的默认值，出现报错。

### DISCUSSION THREAD

**Discussion by zenghaiy** [zouyaoji/vue-cesium#542] [Q&A] [answered]
![code](https://github.com/zouyaoji/vue-cesium/assets/88429575/23de0578-a55a-4f7c-b3b8-324f53b6ecfd)
![error](https://github.com/zouyaoji/vue-cesium/assets/88429575/c4a8f424-0936-47bd-a6b6-360b0fad23b6)

**Comment by zouyaoji** [comment, marked as answer]
是这样哈，vue-cesium加载Cesium.js实际上是动态script标签加上去的，所以按你代码在setup中使用Cesium说找不到，因为还没初始化好呢。3个办法呢。
1、在vc-viewer的cesiumReady事件后去初始化Cesium的原生对象。但Cesium的原生对象，尤其是是一些复杂的对象，不建议用ref初始化，会比较好性能，用shallowRef都要好一点。
2、直接把CesiumJS和它的样式文件在index.html中引入，这样全局就有CesiumJS了。
3、像Cesium.UrlTemplateImageryProvider等已经封装成组件的https://zouyaoji.top/vue-cesium/#/zh-CN/component/providers/vc-imagery-provider-urltemplate  可以直接用现成组件传url，ref只需要写一个字符串即可，性能开销小。

推荐3。
